python/pyiceberg/manifest.py

```diff
@@ -266,11 +267,25 @@ def __init__(self, *data: Any, **named_data: Any) -> None:
 
     def fetch_manifest_entry(self, io: FileIO) -> List[ManifestEntry]:
         file = io.new_input(self.manifest_path)
-        return list(read_manifest_entry(file))
+        return [_inherit_sequence_number(entry, self) for entry in read_manifest_entry(file)]
```

Comment:
Why not do this in `read_manifest_entry`? I think it would make sense to pass the sequence number into that method so that there isn't a way for people to read without sequence numbers.

Reply:
Then we have to pass the `ManifestFile` through, but you're right. In hindsight, the `read_manifest_entry` should have been marked private.

Reply:
Hmm. There were two paths for fetching the manifest entries, both through the `files` and `live_files`. I've removed the `files` and `live_files` to do everything through the ManifestList. Otherwise, we had to change the `files` and `live_files signatures by adding the ManifestList for doing the inheritance.